### PR TITLE
Implement get page ids by the given file id

### PR DESCRIPTION
**Pull request by beinan**
### What changes are proposed in this pull request?

Implement get page ids by the given file id

### Why are the changes needed?

The local cache callers are able to get a list of pages for a particular file, and then they can delete the pages for GDPR or invalidation purpose

### Does this PR introduce any user facing changes?

Add a new api for local cache

### CODE REVIEW

File: core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java [hunk 1]
```diff
@@ -617,6 +619,21 @@ private boolean restoreInternal(Path rootDir, PageStoreOptions options) {
     return true;
   }
 
+  @Override
+  public List<PageId> getPageIdsByFileId(String fileId, long fileLength) {
+    int numOfPages = (int) (fileLength / mPageSize);
+    List<PageId> pageIds = new ArrayList<>(numOfPages);
+    try (LockResource r2 = new LockResource(mMetaLock.readLock())) {
```
Comment (apc999) [inline core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java:626]: why starting from r2 :)

File: core/client/fs/src/main/java/alluxio/client/file/cache/CacheManager.java [hunk 2]
```diff
@@ -208,6 +209,16 @@ default int get(PageId pageId, int pageOffset, int bytesToRead, byte[] buffer,
   int get(PageId pageId, int pageOffset, int bytesToRead, byte[] buffer, int offsetInBuffer,
       CacheContext cacheContext);
 
+  /**
+   * Get page ids by the given file id.
+   * @param fileId file identifier
+   * @param fileLength file length
+   * @return a list of page ids which belongs to the file
+   */
+  default List<PageId> getPageIdsByFileId(String fileId, long fileLength) {
```
Comment (apc999) [inline core/client/fs/src/main/java/alluxio/client/file/cache/CacheManager.java:218]: can you annotate somewhere in Javadoc, e.g. stating that fileLength will not be needed after we have per-file metadata?

File: core/client/fs/src/main/java/alluxio/client/file/cache/CacheManager.java [hunk 3]
```diff
@@ -208,6 +209,16 @@ default int get(PageId pageId, int pageOffset, int bytesToRead, byte[] buffer,
   int get(PageId pageId, int pageOffset, int bytesToRead, byte[] buffer, int offsetInBuffer,
       CacheContext cacheContext);
 
+  /**
+   * Get page ids by the given file id.
```
Comment (apc999) [inline core/client/fs/src/main/java/alluxio/client/file/cache/CacheManager.java:213]: is it "get page ids" or "get ids of the pages which are cached"?
if later case, shall we called `getCachedPageIdsByFileId`?

**Review by apc999 — APPROVED** [review]

File: core/client/fs/src/main/java/alluxio/client/file/cache/CacheManager.java (hunk 2 above)
Comment (beinan) [reply core/client/fs/src/main/java/alluxio/client/file/cache/CacheManager.java:218]: done

File: core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java (hunk 1 above)
Comment (beinan) [reply core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java:626]: changed it to r

File: core/client/fs/src/main/java/alluxio/client/file/cache/CacheManager.java (hunk 3 above)
Comment (beinan) [reply core/client/fs/src/main/java/alluxio/client/file/cache/CacheManager.java:213]: getCachedPageIdsByFileId looks better, thx!

**Comment by HelloHorizon** [comment]
alluxio-bot, merge this please

**Pull request merged**